lib/vagrant/action/builtin/set_hostname.rb

```diff
@@ -16,7 +16,8 @@ def call(env)
           @app.call(env)
 
           hostname = env[:machine].config.vm.hostname
-          if !hostname.nil?
+          disable_hosts_modification = env[:machine].config.vm.disable_hosts_modification
+          if !hostname.nil? && !disable_hosts_modification
```

Comment:
Looks good! Only suggestion I have is add a `@logger` in the initialize method, and then log a simple `info` level message saying that hosts modification has been disabled and will be skipped since `disable_hosts_modification` is true.